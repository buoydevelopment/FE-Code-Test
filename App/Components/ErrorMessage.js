import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {
  Image,
  Text,
  TouchableOpacity,
  View,
} from 'react-native'
import styles from './Styles/ErrorMessageStyles'


class ErrorMessage extends Component {

  render () {
    const {message} = this.props;
    return (
      <View style={[styles.messageContainer, this.props.styles]}>
        <Text style={styles.text}>{message ? message : 'An error has occurred.'}</Text>
      </View>
    )
  }

  static propTypes = {
    message: PropTypes.string,
  };

}

export default ErrorMessage
