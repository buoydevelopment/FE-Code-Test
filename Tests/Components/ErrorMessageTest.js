import 'react-native'
import React from 'react';
import ErrorMessage from '../../App/Components/ErrorMessage';//Import component to test

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<ErrorMessage message="The connection to the server has failed."/>).toJSON();//Pass the component to test to the create function
  expect(tree).toMatchSnapshot();
});

test('renders correctly with a default message', () => {
  const tree = renderer.create(<ErrorMessage />).toJSON();//Pass the component to test to the create function
  expect(tree).toMatchSnapshot();
});
