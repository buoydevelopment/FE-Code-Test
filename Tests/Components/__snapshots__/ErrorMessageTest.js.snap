// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#3b5998",
        "borderRadius": 10,
        "elevation": 3,
        "flex": 1,
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "HelveticaNeue-Italic",
        "fontSize": 16,
        "maxWidth": 150,
        "textAlign": "center",
      }
    }
  >
    The connection to the server has failed.
  </Text>
</View>
`;

exports[`renders correctly with a default message 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#3b5998",
        "borderRadius": 10,
        "elevation": 3,
        "flex": 1,
        "flexDirection": "row",
        "height": 120,
        "justifyContent": "center",
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontFamily": "HelveticaNeue-Italic",
        "fontSize": 16,
        "maxWidth": 150,
        "textAlign": "center",
      }
    }
  >
    An error has occurred.
  </Text>
</View>
`;
